moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -279,8 +279,14 @@ void ompl_interface::ModelBasedPlanningContext::useConfig()
     return;
   std::map<std::string, std::string> cfg = config;
 
+  auto it = cfg.find("debug_mode");
+  if (it != cfg.end() && !(it->second.compare("1")))
+  {
+    ompl_simple_setup_->print();
```

Comment:
```suggestion
  std::stringstream debug_out;
  ompl_simple_setup_->print(debug_out);
  RCLCPP_DEBUG(LOGGER, "%s", rclcpp::get_c_string(debug_out.str()));
```

Reply:
This is probably my ignorance, but I don't see any output from this. Am I doing something wrong?

`ros2 launch moveit_resources_panda_moveit_config demo.launch.py --debug`

Reply:
It makes sense to call print right after `SimpleSetup::solve` is called (maybe in [ompl_interface::ModelBasedPlanningContext::postSolve](https://github.com/ros-planning/moveit2/blob/main/moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp#L742-L753)?). I modified a basic demo program print the SimpleSetup object right after it's created and before it's configured. In that case the output would look something like this:
```
Properties of the state space 'SE2CompoundSpace0'
  - signature: 6 4 3 1 2 2 1 
  - dimension: 3
  - extent: 4.39922
Call setup() before to get more information
Settings for the state space 'SE2CompoundSpace0'
  - state validity check resolution: 1%
  - valid segment count factor: 1
  - state space:
Compound state space 'SE2CompoundSpace0' of dimension 3 (locked) [
Real vector state space 'RealVectorSpace1' of dimension 2 with bounds: 
  - min: -1 -1 
  - max: 1 1 
 of weight 1
SO2 state space 'SO2Space2'
 of weight 0.5
]
No registered projections

Declared parameters:
longest_valid_segment_fraction = 0.01
valid_segment_count_factor = 1
Valid state sampler named uniform with parameters:
nr_attempts = 100
Start states:
Goal = nullptr
OptimizationObjective = nullptr
There are 0 solutions
```
If I call SimpleSetup::print right after planning, the output would look like this:
```
Properties of the state space 'SE2CompoundSpace0'
  - signature: 6 4 3 1 2 2 1 
  - dimension: 3
  - extent: 4.39922
  - sanity checks for state space passed
  - probability of valid states: 0.808
  - average length of a valid motion: 1.71476
  - average number of samples drawn per second: sampleUniform()=1.81818e+07 sampleUniformNear()=2.38095e+07 sampleGaussian()=7.0922e+06
Settings for the state space 'SE2CompoundSpace0'
  - state validity check resolution: 1%
  - valid segment count factor: 1
  - state space:
Compound state space 'SE2CompoundSpace0' of dimension 3 (locked) [
Real vector state space 'RealVectorSpace1' of dimension 2 with bounds: 
  - min: -1 -1 
  - max: 1 1 
 of weight 1
SO2 state space 'SO2Space2'
 of weight 0.5
]
Registered projections:
  - <default>
Projection of dimension 2
Cell sizes (computed defaults): [0.1 0.1]

Declared parameters:
longest_valid_segment_fraction = 0.01
projection.cellsize.0 = 0.1
projection.cellsize.1 = 0.1
projection.cellsize_factor = 0
valid_segment_count_factor = 1
Valid state sampler named uniform with parameters:
nr_attempts = 100
Planner LBKPIECE1 specs:
Multithreaded:                 No
Reports approximate solutions: No
Can optimize solutions:        No
Aware of the following parameters: border_fraction min_valid_path_fraction range
Declared parameters for planner LBKPIECE1:
border_fraction = 0.9
min_valid_path_fraction = 0.5
range = 0.879845
Start states:
Compound state [
RealVectorState [0.5 -0.35144]
SO2State [-1.25508]
]
Goal state, threshold = 2.22045e-16, memory address = 0x600001cad3c8, state = 
Compound state [
RealVectorState [-0.5 0.389726]
SO2State [-1.62618]
]
OptimizationObjective = nullptr
There are 1 solutions
```
The print method is defined [here](https://github.com/ompl/ompl/blob/main/src/ompl/geometric/src/SimpleSetup.cpp#L226-L240). I find it useful to see the exact planner that was being used, including its parameters, as well as the problem definition (which includes the start and goal states, any optimization objectives, and how many solutions were found). It's a useful sanity check to make sure parameters were read correctly and planning is performed as expected.

Reply:
> This is probably my ignorance, but I don't see any output from this. Am I doing something wrong?
> 
> `ros2 launch moveit_resources_panda_moveit_config demo.launch.py --debug`

You need to set the log level for the node: https://docs.ros.org/en/rolling/Tutorials/Logging-and-logger-configuration.html#logger-level-configuration-command-line. Note, that the example is a run command, for launch files you'd have to add the `--log-level` arg as a node argument. There also is a service for setting logger levels dynamically for a node namespace, I haven't tried that one yet, though. https://docs.ros.org/en/rolling/Tutorials/Logging-and-logger-configuration.html#logger-level-configuration-externally.

@mamoll thanks for the examples. We should definitely print after configuration or solve, the earlier prints don't even show the planner alg.

Reply:
OK, I changed the PR to print at the DEBUG level and moved the print to `postSolve()`.